File: app/src/main/java/me/aluceps/sandbox/di/FragmentComponent.java
package me.aluceps.sandbox.di;

import dagger.Subcomponent;
import me.aluceps.sandbox.di.scope.FragmentScope;
import me.aluceps.sandbox.view.main.MainFragment;

@FragmentScope
@Subcomponent(modules = FragmentModule.class)
public interface FragmentComponent {

    void inject(MainFragment __);
}

File: app/src/main/java/me/aluceps/sandbox/di/FragmentComponent.kt
package me.aluceps.sandbox.di

import dagger.Subcomponent
import me.aluceps.sandbox.di.scope.FragmentScope
import me.aluceps.sandbox.view.main.MainFragment

@FragmentScope
@Subcomponent(modules = [(FragmentModule::class)])
interface FragmentComponent {

    fun inject(fragment: MainFragment)
}
